code/data_structures/src/list/singly_linked_list/operations/find/search_element_python.py

```diff
@@ -0,0 +1,80 @@
+# Python Utility to add element at the end of the Singly Linked List
+# Part of Cosmos by OpenGenus Foundation
+
+
+class Node:
+    '''
+    A Node in a singly-linked list
+    Parameters
+    -------------------
+    data :
+        The data to be stored in the node
+    next:
+        The link to the next node in the singly-linked list
+    '''
+
+    def __init__(self, data=None, next=None):
+        ''' Initializes node structure'''
+        self.data = data
+        self.next = next
+
+    def __repr__(self):
+        ''' Node representation as required'''
+        return (self.data)
+
+
+class SinglyLinkedList:
+    '''
+    A structure of singly linked lists
+    '''
+
+    def __init__(self):
+        '''Creates a Singly Linked List in O(1) Time'''
+        self.head = None
+
+    def append(self, data):
+        '''
+        Inserts New data at the ending of the Linked List
+        Takes O(n) time
+        '''
+        if not self.head:
+            self.head = Node(data=data)
+            return
+
+        current_node = self.head
+        while current_node.next:
+            current_node = current_node.next
+
+        new_node = Node(data=data, next=None)
+        current_node.next = new_node
+
+    def find(self, data):
+        if self.head is None:
```

Comment:
I think this would not be the proper way to implement a find function. In the usual case, _just_ the index of the _first_ instance of the found element is returned, and otherwise -1 (an invalid index). This would also shorten the code significantly to just a simple loop. Also, please mention like in related comments that this is an O(n) operation.